pkg/clusters/types/gke/builder.go

```diff
@@ -205,6 +217,12 @@ func (b *Builder) Build(ctx context.Context) (clusters.Cluster, error) {
 			Channel: channel,
 		}
 	}
+	if b.enableNetworkPolicy {
+		pbcluster.NetworkPolicy = &containerpb.NetworkPolicy{
+			Enabled:  true,
+			Provider: containerpb.NetworkPolicy_CALICO,
```

Comment:
While currently it seems that only CALICO is allowed here (ref: https://pkg.go.dev/cloud.google.com/go/container@v1.42.0/apiv1/containerpb#NetworkPolicy_Provider) how about making the flag not a bool (and set Calico as the implementation of choice) but rather make the flag accept a string and validate it against the allowed set of implementations?

This way in the future we'll be able to add more options to choose as the implementation. WDYT?

Reply:
Well, I guess they'll never support a second network policy provider in standard clusters, so by using a boolean can simplify the API here.  
They've created Dataplane v2 which has network policy support built-in, and in auto-pilot clusters, they use Dataplane v2 by default. So I guess their standard clusters are going to be sort of "legacy" in the near future.

Reply:
Source: https://cloud.google.com/kubernetes-engine/docs/how-to/network-policy#calico_migrate

Reply:
Fair enough. Thanks for the context and the link 👍 

I just wanted to make sure that we're not locking ourselves in for the future. 👍